lldb/include/lldb/ValueObject/DILAST.h

```diff
@@ -81,6 +90,27 @@ class IdentifierNode : public ASTNode {
   std::string m_name;
 };
 
+class UnaryOpNode : public ASTNode {
+public:
+  UnaryOpNode(uint32_t location, UnaryOpKind kind, ASTNodeUP rhs)
+      : ASTNode(location, NodeKind::eUnaryOpNode), m_kind(kind),
+        m_rhs(std::move(rhs)) {}
+
+  llvm::Expected<lldb::ValueObjectSP> Accept(Visitor *v) const override;
+  bool is_rvalue() const override { return m_kind != UnaryOpKind::Deref; }
+
+  UnaryOpKind kind() const { return m_kind; }
+  ASTNode *rhs() const { return m_rhs.get(); }
+
+  static bool classof(const ASTNode *node) {
+    return node->GetKind() == NodeKind::eUnaryOpNode;
+  }
+
+private:
+  UnaryOpKind m_kind;
+  ASTNodeUP m_rhs;
```

Comment:
It's kinda weird to have the *right* side of a unary expression (and it would get particularly weird if we end up having a postfix `++` operator). Clang calls this `Val` or SubExpr`, so maybe call this m_(sub_)expr?